cgp/node_validation.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+
+try:
+    import sympy  # noqa: F401
+    from sympy.core import expr as sympy_expr  # noqa: F401
+
+    sympy_available = True
+except ModuleNotFoundError:
+    sympy_available = False
+
+try:
+    import torch  # noqa: F401
+
+    torch_available = True
+except ModuleNotFoundError:
+    torch_available = False
+
+from typing import Type, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .node import NAryOperatorNode  # noqa: F401
+    from .genome import Genome  # noqa: F401
+
+
+def _create_genome(cls: Type["NAryOperatorNode"]) -> "Genome":
+    # delayed imports to avoid circular imports
+    from .genome import Genome, ID_INPUT_NODE, ID_OUTPUT_NODE, ID_NON_CODING_GENE
+
+    primitives = (cls,)
+    genome = Genome(1, 1, 1, 1, 1, primitives)
+    dna = [ID_INPUT_NODE]
+    arity = max(cls._arity, 1)
+    for _ in range(arity):
+        dna += [ID_NON_CODING_GENE]
+    dna += [0]
+    for _ in range(arity):
+        dna += [0]
+    dna += [ID_OUTPUT_NODE, 1]
+    for _ in range(arity - 1):
+        dna += [ID_NON_CODING_GENE]
+    genome.dna = dna
+
+    return genome
+
+
+def check_to_func(cls: Type["NAryOperatorNode"]) -> None:
+    # delayed imports to avoid circular imports
+    from .cartesian_graph import CartesianGraph
+
+    genome = _create_genome(cls)
+
+    f = CartesianGraph(genome).to_func()
+    x = [1.5]
+    f(x)[0]
+
+
+def check_to_numpy(cls: Type["NAryOperatorNode"]) -> None:
+    # delayed imports to avoid circular imports
+    from .cartesian_graph import CartesianGraph
+
+    genome = _create_genome(cls)
+
+    f = CartesianGraph(genome).to_numpy()
+    x = np.ones((3, 1)) * 1.5
```

Comment:
any reason for the magic 1.5?

Reply:
hmm, no, it's a pretty number, no? i can use `3.14159`, if that's more obvious ;)

Reply:
76015a61a

Reply:
why not just `1.0`, i.e., no number at all?

Reply:
I think this is more maintainable because it does not raise questions like mine.

Reply:
i generally don't like using `1.0` since it can lead to false positives: for example imagine a test that should use a constant node (which by default has output 1.0) but mistakenly forwards the input, or a test that should use multiplication but mistakenly uses division, and so on. by using "non-trivial" values we can avoid some of these problems. maybe we should introduce a `magic_constant` in conftest with such an explanation?

Reply:
Okay, that makes sense, thanks for the explanation. Are we using this in other places, too? If yes, than a fixture would be a good idea. If not, just adding a short explanation here would suffice, I think.